### PR TITLE
feat: use input-neutral verbs

### LINKED ISSUE #18504: Use generic verbs that work with any input method in docs
### Intro

I tried my hand at changing both the docs and the code, and got stuck on the code part. So I'm splitting this work into two parts:

- #18504
- #19164

I'll do the docs changes myself, and let one of the maintainers fix-up the code.

### Describe the proposed change(s).

I want to change our docs so that we use generic verbs that work with any input method.

The Microsoft style guide has a page that explains "describing interactions with UI": [^quote]

> Customers interact with products using different input methods: keyboard, mouse, touch, voice, and more. So use generic verbs that work with any input method. Don't use input-specific verbs, such as _click_ or _swipe_. Instead, use the verbs listed here.

Their style guide has a table with good verbs, and examples on how to use them in a sentence.

I've searched our repository for some common input verbs:

| Input specific verb | Hits on `main` |
| --- | --- |
| Click | 109 |
| Press | 1 |
| Swipe | No relevant hits |
| Touch | 2 relevant hits |
| Tick | 8 |
| Ticking | 2 |

## Todo in docs:

- [x] Replace `click` and `clicking` with `select`
- [x] Replace `press` and `pressing`
- [x] Replace `swipe`, no hits
- [x] Replace `touch`
- [x] Replace `tick` with `select`
- [x] Replace `ticking` with `selecting`

[^quote]: https://learn.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Replace words that refer to input methods/devices with general verbs

## Context

- Closes #18504.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/utils/read-only-issue-body.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@ import { regEx } from '../../../util/regex';
...
 export function readOnlyIssueBody(body: string): string {
   return body
-    .replace(' only once you click their checkbox below', '')
-    .replace(' unless you click a checkbox below', '')
-    .replace(' To discard all commits and start over, click on a checkbox.', '')
-    .replace(regEx(/ Click (?:on |)a checkbox.*\./g), '')
+    .replace(' only once you select their checkbox below', '')
+    .replace(' unless you select a checkbox below', '')
+    .replace(' To discard all commits and start over, select a checkbox.', '')
+    .replace(regEx(/ Select (?:on |)a checkbox.*\./g), '')
```
Comment (HonkingGoose) [inline lib/modules/platform/utils/read-only-issue-body.ts:8]: I guess we need to change this line of code, as we should never say:

> Select **on** a checkbox.

File: lib/modules/platform/utils/read-only-issue-body.spec.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ describe('modules/platform/utils/read-only-issue-body', () => {
     it('removes all checkbox-related instructions', () => {
       expect(readOnlyIssueBody(issueBody)).toEqual(
         expect.not.stringMatching(
-          /click (?:(?:on |)a|their) checkbox|check the box below/gi
+          /select (?:(?:on |)a|their) checkbox|check the box below/gi
```
Comment (HonkingGoose) [inline lib/modules/platform/utils/read-only-issue-body.spec.ts:17]: This line needs to be changed as well, I think.

File: lib/modules/manager/maven/index.spec.ts [hunk 3]
```diff
@@ -229,7 +229,7 @@ describe('modules/manager/maven/index', () => {
       expect(packages).toMatchSnapshot();
     });
 
-    it('should not touch content if new and old versions are equal', () => {
+    it('should not edit content if new and old versions are equal', () => {
```
Comment (HonkingGoose) [inline lib/modules/manager/maven/index.spec.ts:232]: This is a bit of a dubious change, I think _edit_ is the correct verb here. But I'm not familiar with the code, so please review this line. 😉

**Review by HonkingGoose — COMMENTED** [review]
I've basically brute-forced my way into these changes, by finding/replacing the strings in the tests, docs and code.

The code needs more changes, but I don't feel confident editing the code fully. Can you guys suggest the correct code?

---

Edit: I see another test is failing as well, that's about truncating the code. I'll need your help to fix that as well. 🙈

File: lib/modules/platform/utils/read-only-issue-body.ts [hunk 4]
```diff
@@ -2,10 +2,10 @@ import { regEx } from '../../../util/regex';
 
 export function readOnlyIssueBody(body: string): string {
   return body
-    .replace(' only once you click their checkbox below', '')
-    .replace(' unless you click a checkbox below', '')
-    .replace(' To discard all commits and start over, click on a checkbox.', '')
-    .replace(regEx(/ Click (?:on |)a checkbox.*\./g), '')
+    .replace(' only once you select their checkbox below', '')
```
Comment (HonkingGoose) [inline lib/modules/platform/utils/read-only-issue-body.ts:5]: Am I making a **breaking change** here, by accident? What happens to old dashboards that still use the verb `click`?

File: lib/modules/platform/utils/read-only-issue-body.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/platform/utils/read-only-issue-body.ts:8]: "Select a checkbox"?

Comment (HonkingGoose) [reply lib/modules/platform/utils/read-only-issue-body.ts:8]: Want me to change it to "Select **this** checkbox" then?

File: lib/modules/platform/utils/read-only-issue-body.ts (hunk 4 above)
Comment (rarkins) [reply lib/modules/platform/utils/read-only-issue-body.ts:5]: They should be updated in place

File: lib/workers/repository/update/pr/body/controls.spec.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      branchConfig.branchName = 'branchName';
+    });
+
+    describe(`when the branch is modified`, () => {
+      beforeEach(() => {
+        git.isBranchModified.mockResolvedValue(true);
+      });
+
+      it('has the correct contents', async () => {
+        expect(await getControls(branchConfig)).toContain(
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/body/controls.spec.ts:24]: ESLint is failing here:

> Unexpected `await` of a non-Promise (non-"Thenable") value.

File: lib/workers/repository/update/pr/body/controls.spec.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      });
+    });
+
+    describe(`when the branch is not modified`, () => {
+      beforeEach(() => {
+        git.isBranchModified.mockResolvedValue(false);
+      });
+
+      it('has the correct contents', async () => {
+        expect(await getControls(branchConfig)).toContain(
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/body/controls.spec.ts:40]: ESLint is failing here as well:

> Unexpected `await` of a non-Promise (non-"Thenable") value.

**Review by HonkingGoose — COMMENTED** [review]
I fixed the Jest snapshot error by updating the snapshot. However ESLint is still complaining.

Also I still need help with the code changes, as not all code is ready to go. 😉

File: lib/modules/platform/utils/read-only-issue-body.spec.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/platform/utils/read-only-issue-body.spec.ts:17]: this needs to be more fixed to match the changed wording in file below

**Comment by HonkingGoose** [comment]
I'm thinking of splitting the work into two parts:

- Docs only changes to do by me, @HonkingGoose
- Code changes + test changes: to do by maintainer

I don't want to merge this PR as-is, because there's still more code to fix. But I can't really fix the code myself.

**Pull request closed without merging**